### PR TITLE
Implemented alternative date/time parsing options for json

### LINKED ISSUE #773: Parse temporal and decimal types in Debezium-encoded JSON
Debezium uses different encodings for 
[temporal](https://debezium.io/documentation/reference/stable/connectors/mysql.html#mysql-temporal-types) and [decimal](https://debezium.io/documentation/reference/stable/connectors/mysql.html#mysql-decimal-types) types depending on (1) connector config, (2) type precision specified in SQL, (3) source database.

Todos:
- Figure out a way to detect the encoding (@ryzhyk )
  - I gave up on this one. There really isn't a good solution when using JSON.
- [x] Implement support for parsing these types in the serde-based framework (@ryzhyk ): #816 , #812 
- [ ] Implement support for parsing these types in JIT (@Kixiron): #835

### PR DESCRIPTION
**Pull request by Kixiron**
Is this a user-visible change (yes/no): no

Added the ability to parse dates/times from more formats, fixes #773
Changes config format, columns should look like this now:
```json
{
    "key": "/bar",
    "spec": null,
}

{
    "key": "/bar",
    "spec": {
        "DateTimeFromStr": {
            "format": "%F"
        }
    }
}

{
    "key": "/bar",
    "spec": "TimeFromMicros",
}

{
    "key": "/bar",
    "spec": "TimeFromMillis",
}

{
    "key": "/bar",
    "spec": "DateFromDays",
}
```

### CODE REVIEW

**Comment by ryzhyk** [comment]
@mihaibudiu, I'm guessing all you need to do is change `DateTime` to `DateTimeFromStr` in your JIT IR backend.

**Comment by mihaibudiu** [comment]
I will push the compiler changes to this branch directly

File: crates/dataflow-jit/src/codegen/intrinsics/deserialize.rs [hunk 1]
```diff
@@ -196,6 +196,26 @@ pub(super) extern "C" fn deserialize_json_date(
...
+) -> bool {
+    // The json pointer we're accessing the map with
+    let json_pointer = unsafe { str_from_raw_parts(json_pointer_ptr, json_pointer_len) };
+
+    if let Some(days) = map.pointer(json_pointer).and_then(Value::as_i64) {
+        // TODO: This silently truncates, do we want that?
+        place.write(days as i32);
+        false
+
+    // Otherwise the value couldn't be found and is considered null
```
Comment (mihaibudiu) [inline crates/dataflow-jit/src/codegen/intrinsics/deserialize.rs:213]: this comment (and the subsequent ones) should probably be in the else branch., with a different indentation. the empty line is also unnecessary

File: crates/dataflow-jit/src/codegen/json/deserialize.rs [hunk 2]
```diff
@@ -252,30 +252,130 @@ impl Codegen {
...
+                                );
+
+                                builder.call_fn(
+                                    deserialize,
+                                    &[column_place, json_pointer, json_pointer_len, json_map],
+                                )
+                            }
+
+                            JsonColumnParseSpec::TimeFromMicros
+                            | JsonColumnParseSpec::TimeFromMillis => unreachable!(),
```
Comment (mihaibudiu) [inline crates/dataflow-jit/src/codegen/json/deserialize.rs:298]: an error message would make it much easier to diagnose misconfigurations.
e.g., "configuration X not supported for type Y"

**Review by mihaibudiu — APPROVED** [review]

File: crates/dataflow-jit/src/codegen/intrinsics/deserialize.rs (hunk 1 above)
Comment (Kixiron) [reply crates/dataflow-jit/src/codegen/intrinsics/deserialize.rs:213]: The comment is on the else branch

**Pull request merged**